src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 200) {
+    this.health = health;
+    this.name = 'Animal';
+  }
+
+  dead(healthPoints) {
+    this.health += healthPoints;
+
+    if (this.health < 100) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health);
+    }
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, 100) {
+    this.hidden = 'hidden';
+  }
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(mammal) {
+    if (
+      !mammal.hidden
+    ) {
+      mammal.dead(50 * 0);
+    }
```

Comment:
The `bite` method should check if the `mammal` is an instance of `Herbivore` and not hidden before applying damage. Also, the damage calculation is incorrect due to `50 * 0`; it should simply decrease the mammal's health by `50`. Moreover, the method should not work if the mammal is another carnivore. Add the necessary checks and correct the damage calculation.